### DISCUSSION TITLE
Dubbio su caddy e duckdns

### DISCUSSION THREAD

**Discussion by federoco** [indomus/forum#2191] [GENERALE - Discussioni generiche]
Salve a tutti, ho installato HA tramite Docker con Docker Compose su Raspberry; ho cablato tutta casa e le varie prese etherner nelle stanze sono gestire tramite lo switch Cisco CISCO SG220-26P, il quale a sua volta è connesso al modem FritzBox 7530.
Lo switch è gestito in DHCP dal FritzBox, mentre ho riservato al Raspberry un IP fisso, sempre tramite il FritzBox.
Sto seguendo i primi passaggi della guida "Collegarsi da remoto a Home Assistant via Reverse Proxy".
Ovvero sto installando Caddy (https://indomus.it/guide/come-installare-e-configurare-caddy-con-docker-su-raspbian-di-raspberry-pi/)

Ho ripetuto varie volte i passaggi della guida ma ogni volta che provo a collegarmi da remoto ottengo un 404 Bad request.
Sto cercando di chiarire ogni passaggio.
Il dubbio riguarda il paragrafo "Aggiornamento di DuckDNS" dove si dice che non è necessario creare un container ad hoc, se - appunto - abbiamo un dispositivo che si occupa automaticamente di comunicare la variazione di Ip a Duckdns.
Io ho configurato DynDNS sul FritzBox (Internet --> Abilitazioni --> DynDNS) secondo le istruzioni che fornisce Duckdns. 
Mi confermate che non è necessario creare il container su Docker?

Fare chiarezza su questo passaggio mi è utile prima di ripetere da capo tutta la guida.
Grazie.

**Comment by snakuzzo** [comment]
L'errore 404 Bad Request non esiste...credo sia 400 Bad Request.
Se l'errore è questo, ti sei perso l'ultimo paragrafo della guida "RICONFIGURARE HOME ASSISTANT", che ti rimanda [QUI](https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reverse-proxy/#homeassistant)

Quanto alla seconda domanda, non ho capito se con "DynDNS" intendi il nome dell'utility presente sul fritzbox oppure il servizio dyndns.it o dyndns.com. Sul fritzbox "FORSE" (non ricordo, non avendolo più) puoi impostare un servizio non in lista impostando manualmente FQDN, username e password (dove andrai a mettere il token di duckdns)

**Comment by federoco** [comment]
Ciao Diego, ti ringrazio moltissimo per la tua risposta pressochè immediata. 
Hai ragione, l'errore è 400 Bad request.

Dopo aver seguito di nuovo la guida, sono riuscito a far funzionare il sistema.

Vorrei suggerire, nella guida relativa a Caddy, di separare le indicazioni per chi usa solo Docker o per chi si serve di Docker Compose, perchè è facile fare confusione fra i due sistemi.

Colgo l'occasione per fare a te e tutta la redazione di Indomus auguri per un sereno Natale!

**Reply by snakuzzo** [reply]
Non ho capito l'indicazione per docker e docker-compose.
Sempre di Docker parliamo...non cambia niente. docker-compose è semplicemente un utility che ti permette di fare "delle cose". Tutto qua.
I concetti sono esattamente gli stessi.
